lib-injection/sitecustomize.py

```diff
@@ -2,65 +2,238 @@
 This module when included on the PYTHONPATH will update the PYTHONPATH to point to a directory
 containing the ddtrace package compatible with the current Python version and platform.
 """
+
+from __future__ import print_function  # noqa: E402
+
+from collections import namedtuple
+import csv
+import json
 import os
+import platform
+import re
+import subprocess
 import sys
 import time
+from typing import Tuple
+
+
+Version = namedtuple("Version", ["version", "constraint"])
+
+
+def parse_version(version: str) -> Tuple:
+    constraint_idx = re.search(r"\d", version).start()
+    numeric = version[constraint_idx:]
+    constraint = version[:constraint_idx]
+    parsed_version = tuple(int(re.sub("[^0-9]", "", p)) for p in numeric.split("."))
+    return Version(parsed_version, constraint)
+
+
+RUNTIMES_ALLOW_LIST = {
+    "cpython": {"min": parse_version("3.7"), "max": parse_version("3.12")},
+}
+
+FORCE_INJECT = os.environ.get("DD_INJECT_FORCE", "").lower() in (
+    "true",
+    "1",
+    "t",
+)
+FORWARDER_EXECUTABLE = os.environ.get("DD_TELEMETRY_FORWARDER_PATH")
+TELEMETRY_ENABLED = os.environ.get("DD_INJECTION_ENABLED")
+DEBUG_MODE = os.environ.get("DD_TRACE_DEBUG", "").lower() in ("true", "1", "t")
+INSTALLED_PACKAGES = None
+PYTHON_VERSION = None
+PYTHON_RUNTIME = None
+PKGS_ALLOW_LIST = None
+
+
+def build_installed_pkgs():
+    if sys.version_info >= (3, 8):
+        from importlib import metadata as importlib_metadata
+
+        installed_packages = {pkg.metadata["Name"]: pkg.version for pkg in importlib_metadata.distributions()}
+    else:
+        import pkg_resources
```

Comment:
`pkg_resources` might not be installed, this will get caught by the main `try/catch` around `_inject`, but we are cool with it failing if we cannot check packages versions?

Reply:
we can also try `import importlib_metadata` if they are on Python 3.7. if they happen to have it installed, will be the same as what we get from `importlib.metadata`.